tests/integration/targets/mso_tenant_site/tasks/main.yml

```diff
@@ -0,0 +1,661 @@
+  fail:
+    msg: 'Please define the following variables: mso_hostname, mso_username and mso_password.'
+  when: mso_hostname is not defined or mso_username is not defined or mso_password is not defined
+
+
+# CLEAN ENVIRONMENT
+- name: Set vars
+  set_fact: 
+    mso_info: &mso_info
+      host: '{{ mso_hostname }}'
+      username: '{{ mso_username }}'
+      password: '{{ mso_password }}'
+      validate_certs: '{{ mso_validate_certs | default(false) }}'
+      use_ssl: '{{ mso_use_ssl | default(true) }}'
+      use_proxy: '{{ mso_use_proxy | default(true) }}'
+      output_level: '{{ mso_output_level | default("info") }}'
+
+- name: Remove schemas
+  mso_schema:
+    <<: *mso_info
+    schema: '{{ item }}'
+    state: absent
+  loop:
+  - '{{ mso_schema | default("ansible_test") }}_2'
+  - '{{ mso_schema | default("ansible_test") }}'
+
+- name: Dissociate clouds that are associated with ansible_tenant
+  mso_tenant_site:
+    <<: *mso_info
+    tenant: ansible_test
+    site: '{{ item }}'
+    state: absent
+  loop:
+  - '{{ mso_site | default("ansible_test") }}'
+  - '{{ mso_site | default("aws_ansible_test") }}'
+  - '{{ mso_site | default("azure_ansible_test") }}'
+  ignore_errors: true
+
+- name: Remove tenant ansible_test
+  mso_tenant: 
+    <<: *mso_info
+    tenant: ansible_test
+    users:
+      - '{{ mso_username }}'
+    state: absent
+
+- name: Ensure non-cloud site exists
+  mso_site:
+    <<: *mso_info
+    site: '{{ mso_site | default("ansible_test") }}'
+    apic_username: '{{ apic_username }}'
+    apic_password: '{{ apic_password }}'
+    apic_site_id: '{{ apic_site_id }}'
+    urls:
+    - https://{{ apic_hostname }}
+    state: present
+
+- name: Ensure azure site exists
+  mso_site:
+    <<: *mso_info
+    site: '{{ mso_site | default("azure_ansible_test") }}'
+    apic_username: '{{ azure_apic_username }}'
+    apic_password: '{{ azure_apic_password }}'
+    apic_site_id: '{{ azure_site_id }}'
+    urls:
+    - https://{{ azure_apic_hostname }}
+    state: present
+
+- name: Ensure aws site exists
+  mso_site:
+    <<: *mso_info
+    site: '{{ mso_site | default("aws_ansible_test") }}'
+    apic_username: '{{ aws_apic_username }}'
+    apic_password: '{{ aws_apic_password }}'
+    apic_site_id: '{{ aws_site_id }}'
+    urls:
+    - https://{{ aws_apic_hostname }}
+    state: present
+
+- name: Ensure tenant ansible_test exists
+  mso_tenant: 
+    <<: *mso_info
+    tenant: ansible_test
+    users:
+      - '{{ mso_username }}'
+    state: present
+
+- name: Associate non-cloud site with ansible_test in check mode
+  mso_tenant_site:
+    <<: *mso_info
+    tenant: ansible_test
+    site: ansible_test
+    state: present
+  check_mode: yes   
+  register: ncs_cm
+
+- name: Verify ncs_cm
+  assert:
+    that:
+    - ncs_cm is changed
+
+- name: Associate non-cloud site with ansible_test in normal mode
+  mso_tenant_site:
+    <<: *mso_info
+    tenant: ansible_test
+    site: ansible_test
```

Comment:
Site name does not follow variable define in site. If I change it by passing mso_site variable this and all other references will break. Fix all 'site' attribute to reflect site definition with variable like 'aws_{{ mso_site | default("ansible_test") }}'